compose/project.py

```diff
@@ -157,16 +160,19 @@ def get_links(self, service_dict):
     def get_volumes_from(self, service_dict):
         volumes_from = []
         if 'volumes_from' in service_dict:
-            for volume_name in service_dict.get('volumes_from', []):
+            for volume_from_config in service_dict.get('volumes_from', []):
+                volume_from_spec = parse_volume_from_spec(volume_from_config)
+                # Get service
                 try:
-                    service = self.get_service(volume_name)
-                    volumes_from.append(service)
+                    service_name = self.get_service(volume_from_spec.source)
+                    volume_from_spec = VolumeFromSpec(service_name, volume_from_spec.mode)
```

Comment:
@aanand I wasn't sure if ``parse_volume_from_spec`` should return a ``VolumeFromSpec`` as I need to create a new one few lines later..